Update wallpaper for eos6

cassidyjames opened the pull request:
More involved than usual because this is the first time we're adding a dynamic wallpaper (to support the FreeDesktop dark color scheme preference). The gist:

- Adds the new default wallpaper as usual
- Adds a new dark variant of that wallpaper, and installs it
- Adds and installs a new dynamic wallpaper XML file so users can switch back to the default set if they switch away from it (otherwise there's no way to get back to it from the GUI)

Future wallpaper updates will be simpler again, since they'll just require adding the wallpapers and updating the path in meson.

https://phabricator.endlessm.com/T35296

cassidyjames committed:
README: Document how to update the wallpaper

https://phabricator.endlessm.com/T35296

cassidyjames committed:
Wallpaper: Update default for Endless OS 6

https://phabricator.endlessm.com/T35296

cassidyjames committed:
Wallpaper: Add and install dynamic wallpaper

Exposes the dynamic light/dark wallpaper in Settings. Added to the
eos-default-background package since these two wallpapers will be set as
the defaults in GSettings, so it would be annoying to not be able to
switch back to them.

https://phabricator.endlessm.com/T35296

wjt commented on debian/eos-default-background.install:
```diff
@@ -1 +1,3 @@
 usr/share/eos-media/desktop-background-C.jpg
+usr/share/eos-media/desktop-background-dark-C.jpg
+usr/share/gnome-background-properties/endlessos-dynamic.xml
```
Unfortunately this causes an error at package install time because this file is in both packages:

```
dpkg: error processing archive eos-media_1.4.0_all.deb (--install):
 trying to overwrite '/usr/share/gnome-background-properties/endlessos-dynamic.xml', which is also in package eos-default-background 1.4.0
```

I think you can fix this by adjusting `eos-media.install` thusly:

```
-usr/share/gnome-background-properties
+usr/share/gnome-background-properties/endlessos-backgrounds.xml
```

wjt replied:
I tested this as follows:

```
toolbox enter eos-toolbox-latest
dpkg-buildpackage -uc -us -b
exit
cd ..
sudo eos-dev-unlock
sudo dpkg -i eos-default-background_1.4.0_all.deb eos-media_1.4.0_all.deb
```

wjt commented on data/endlessos-dynamic.xml:
```diff
@@ -0,0 +1,14 @@
+<wallpapers>
+  <wallpaper deleted="false">
+    <name>Endless OS Default</name>
+    <filename>/usr/share/eos-media/desktop-background-C.jpg</filename>
+    <filename-dark>/usr/share/eos-media/desktop-background-dark-C.jpg</filename-dark>
```
These two files will also be listed in the `endlessos-backgrounds.xml` file as separate wallpapers, so I think we'll get 3 entries in Settings: one for the combined day/night one, and one each for the individual files. This seems undesirable.

cassidyjames replied:
The way they're presented, I might actually be happy with this; if someone prefers just the light or the dark version, they can explicitly choose to always use it _or_ to use the dynamic version. The dynamic one is distinct since it shows the two versions side-by-side.

That said, it looks like putting them in a sub-folder might prevent them from showing up standalone in the list?? I can test this.

(We want to improve this layout upstream to make it more clear between dynamic and standalone wallpapers, but there's some technical blocker last I checked)

cassidyjames replied:
Can confirm, the individual wallpapers plus the dynamic one show up. I'm okay with this for the aforementioned reason, but I could understand wanting to not have the individual ones displayed, too.

![Screenshot from 2024-04-19 13-25-32](https://github.com/endlessm/eos-media/assets/611168/7238c7a0-a1c0-47ee-9040-6335f2921f8b) | ![Screenshot from 2024-04-19 13-24-18](https://github.com/endlessm/eos-media/assets/611168/6e45807d-8cad-4ac5-933d-2ab6d69cddbb)
--- | ---

wjt commented on README.md:
```diff
@@ -1,4 +1,15 @@
+2. Update the paths to those wallpapers in `default_files/meson.build`
+3. Add licensing information to `default_files/default_images/LICENSE.tsv`
+
+Since the wallpapers are renamed during the build process, the default wallpaper XML does not need to be updated.
```
Strictly speaking they're not renamed but installed twice under different names: once as `usr/share/eos-media/desktop-background-C.jpg` and once as `/usr/share/eos-media/default_images/eos6-mountains.jpg`.

cassidyjames replied:
```suggestion
Since the default wallpapers are also installed with a standardized default file name during the build process, the dynamic wallpaper XML does not need to be updated.
```

cassidyjames committed:
README: more precise caveat about naming in the XML file

cassidyjames committed:
Debian: Explicit path to endlessos-backgrounds.xml

Prevents an error upon installation; before, both packages were trying
to add the same endlessos-dynamic.xml file since the whole directory was
installed by the eos-media package.

https://phabricator.endlessm.com/T35296

cassidyjames committed:
README: add local testing instructions

Thanks to @wjt for sharing the process with me; let's document it so it's easier to find in the future.

https://phabricator.endlessm.com/T35296
